strix/utilities/utils.py

```diff
@@ -520,145 +523,40 @@ def is_bound(data, coord, offsets):
-    return out.to(out_dtype)
-
-
-def draw_segmentation_contour(
-    image: torch.Tensor,
-    masks: torch.Tensor,
-    alpha: float = 0.8,
-    radius: float = 0.2,
-    colors: Optional[List[Union[str, Tuple[int, int, int]]]] = None,
-):
-    """
-    Draws segmentation contour on given RGB image.
-    The values of the input image should be uint8 between 0 and 255.
-
-    Args:
-        image (Tensor): Tensor of shape (3, H, W) and dtype uint8.
-        masks (Tensor): Tensor of shape (num_masks, H, W) or (H, W) and dtype bool.
-        alpha (float): Float number between 0 and 1 denoting the transparency of the masks.
-            0 means full transparency, 1 means no transparency.
-        radius (int): Integer denoting radius of keypoint.
-        colors (list or None): List containing the colors of the masks. The colors can
-            be represented as PIL strings e.g. "red" or "#FF00FF", or as RGB tuples e.g. ``(240, 10, 157)``.
-            When ``masks`` has a single entry of shape (H, W), you can pass a single color instead of a list
-            with one element. By default, random colors are generated for each mask.
-
-    Returns:
-        img (Tensor[C, H, W]): Image Tensor, with segmentation masks drawn on top.
-    """
-    out_dtype = torch.uint8
-    image, masks = __check_image_mask(image, masks)
-    colors_ = __generate_colors(colors, masks.size()[0])
-
-    ndarr = image.permute(1, 2, 0).numpy()
-    img_to_draw = Image.fromarray(ndarr)
-    draw = ImageDraw.Draw(img_to_draw)
-    boundaries = get_bound_2d(masks)
-
-    for i, bound in enumerate(boundaries):
-        for inst_id, pt in enumerate(bound):
-            x1, x2 = pt[2] - radius, pt[2] + radius  # because permute
-            y1, y2 = pt[1] - radius, pt[1] + radius
-            draw.ellipse([x1, y1, x2, y2], fill=tuple(colors_[i]), outline=None, width=0)
-
-    return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=torch.uint8)
-
+    method = 'mask',
+    mask_class_num = 2
+) :
+    # cm = 1/2.54
+    plt.close()
+    ratio = images.shape[-1]/images.shape[-2]
+    fig = plt.figure(figsize=(15 * ratio * ncol, 15 * nrow))
+    axes = fig.subplots(nrow, ncol)
+    color_num  = mask_class_num
+    if color_num >0:
+        colors = get_colors(color_num)
+        cmap = get_colormaps(color_num)
+    for i in range(nrow):
+        for j in range(ncol):
+            axes[i,j].axis('off')
+            if i* ncol+j < images.shape[0]:
+                axes[i,j].imshow(images[i*ncol+j,...].squeeze(), cmap = 'gray', interpolation = 'bicubic')
+                if method == 'mask':
+                    axes[i,j].imshow(np.ma.masked_equal(masks[i*ncol+j,...].squeeze(), 0), cmap, alpha = alpha, norm=matplotlib.colors.Normalize(vmin=1, vmax=color_num))
+                elif method == 'contour':
+                    list = [i for i in np.unique(masks[i*ncol+j,...].squeeze()).tolist() if i != 0 and i is not None]
```

Comment:
equivalent to `[i for i in np.unique(masks[i*ncol+j,...].squeeze()).tolist() if i]`